exec/java-exec/src/test/java/org/apache/drill/TestSelectWithOption.java

```diff
@@ -0,0 +1,203 @@
+        listOf("b", "2")
+      );
+  }
+
+  @Test @Ignore // It does not look like lineDelimiter is working
+  public void testTextLineDelimiter() throws Exception {
+    String tableName = genCSVTable("testTextLineDelimiter",
+        "\"b\"|\"0\"",
+        "\"b\"|\"1\"",
+        "\"b\"|\"2\"");
+
+    testWithResult(format("select columns from table(%s(type => 'TeXT', lineDelimiter => '|'))", tableName),
+        listOf("\"b\""),
+        listOf("\"0\"", "\"b\""),
+        listOf("\"1\"", "\"b\""),
+        listOf("\"2\"")
+      );
+  }
+
+  @Test
+  public void testTextQuote() throws Exception {
+    String tableName = genCSVTable("testTextQuote",
+        "\"b\"|\"0\"",
+        "\"b\"|\"1\"",
+        "\"b\"|\"2\"");
+
+    testWithResult(format("select columns from table(%s(type => 'TeXT', fieldDelimiter => '|', quote => '@'))", tableName),
+        listOf("\"b\"", "\"0\""),
+        listOf("\"b\"", "\"1\""),
+        listOf("\"b\"", "\"2\"")
+        );
+
+    String quoteTableName = genCSVTable("testTextQuote2",
+        "@b@|@0@",
+        "@b$@c@|@1@");
+    // It seems that a parameter can not be called "escape"
+    testWithResult(format("select columns from table(%s(`escape` => '$', type => 'TeXT', fieldDelimiter => '|', quote => '@'))", quoteTableName),
+        listOf("b", "0"),
+        listOf("b$@c", "1") // shouldn't $ be removed here?
+        );
+  }
+
+  @Test
+  public void testTextComment() throws Exception {
+      String commentTableName = genCSVTable("testTextComment",
+          "b|0",
+          "@ this is a comment",
+          "b|1");
+      testWithResult(format("select columns from table(%s(type => 'TeXT', fieldDelimiter => '|', comment => '@'))", commentTableName),
+          listOf("b", "0"),
+          listOf("b", "1")
+          );
+  }
+
+  @Test
+  public void testTextHeader() throws Exception {
+    String headerTableName = genCSVTable("testTextHeader",
+        "b|a",
+        "b|0",
+        "b|1");
+    testWithResult(format("select columns from table(%s(type => 'TeXT', fieldDelimiter => '|', skipFirstLine => true))", headerTableName),
+        listOf("b", "0"),
+        listOf("b", "1")
+        );
+
+    testBuilder()
+        .sqlQuery(format("select a, b from table(%s(type => 'TeXT', fieldDelimiter => '|', extractHeader => true))", headerTableName))
+        .ordered()
+        .baselineColumns("b", "a")
+        .baselineValues("b", "0")
+        .baselineValues("b", "1")
+        .build().run();
+  }
+
+  @Test
+  public void testVariationsCSV() throws Exception {
+    String csvTableName = genCSVTable("testVariationsCSV",
+        "a,b",
+        "c|d");
+    // Using the defaults in TextFormatConfig (the field delimiter is neither "," not "|")
+    String[] csvQueries = {
+//        format("select columns from %s ('TeXT')", csvTableName),
+//        format("select columns from %s('TeXT')", csvTableName),
+        format("select columns from table(%s ('TeXT'))", csvTableName),
```

Comment:
no, but the point was to use a capitalization that is not used anywhere in the code to make sure this was case insensitive. I can vary the spelling but I thought this was sufficient.